### PR TITLE
Specify the different possible values for oidc_provider_token_endpoint 

### PR DESCRIPTION
**Pull request by jvlcek**
The value of oidc_provider_token_endpoint varies based on the configuration and the OpenID-Connect Provider being used.

This PR will describe the different values for oidc_provider_token_endpoint when using Keycloak and IAM.

### CODE REVIEW

**Comment by jvlcek** [comment]
@miq-bot add_label enhancement
@miq-bot assign @abellotti

File: site/_posts/2020-09-03-oidc-auth-api-usage.md [hunk 1]
```diff
@@ -80,13 +85,39 @@ oidc_client_id="my-oidc-client"
...
+```
+
+  * The value of **oidc_provider_token_endpoint** varies based on the configuration and the OpenID-Connect Provider being used. 
+
+  * When using IBM's Identiy Access Management (IAM) for the OpenID-Connect Provider, the value for **oidc_provider_token_endpoint** will be the value of OIDCProviderTokenEndpoint from the OpenID-Connect configuration with **token** replace with **identitytoken**.
+
+  * **e.g.: Using IBM's Identiy Access Management (IAM) with the following OpenID-Connect configuration:**
+
+```bash
+OIDCProviderTokenEndpoint https://cp-console.apps.joevocplusmcmim.cp.fyre.ibm.com/idprovider/v1/auth/token
```
Comment (abellotti) [inline site/_posts/2020-09-03-oidc-auth-api-usage.md:97]: I'd use a sample domain for upstream doc and not a real one. maybe iam.example.com.

File: site/_posts/2020-09-03-oidc-auth-api-usage.md [hunk 2]
```diff
@@ -80,13 +85,39 @@ oidc_client_id="my-oidc-client"
...
+```
+
+  * Then **oidc_provider_token_endpoint** will be:
 
-# OIDCProviderTokenEndpoint from the OpenID-Connect configurationt
-# Sometimes OIDCOauthIntrospectionEndpoint without the trailing "introspect"
+```bash
+oidc_provider_token_endpoint="https://cp-console.apps.joevocplusmcmim.cp.fyre.ibm.com/idprovider/v1/auth/identitytoken"
```
Comment (abellotti) [inline site/_posts/2020-09-03-oidc-auth-api-usage.md:103]: same as above

File: site/_posts/2020-09-03-oidc-auth-api-usage.md [hunk 3]
```diff
@@ -80,13 +85,39 @@ oidc_client_id="my-oidc-client"
...
+```
+
+  * When using Keycloak for the OpenID-Connect Provider the value for **oidc_provider_token_endpoint** will be the value of OIDCOauthIntrospectionEndpoint without the trailing **introspect**
```
Comment (abellotti) [inline site/_posts/2020-09-03-oidc-auth-api-usage.md:106]: OIDCOAuthIntrospectionEndpoint

File: site/_posts/2020-09-03-oidc-auth-api-usage.md [hunk 4]
```diff
@@ -80,13 +85,39 @@ oidc_client_id="my-oidc-client"
 
 # OIDCOauthIntrospectionEndpoint from the OpenID-Connect configuration
 oidc_auth_introspection_endpoint="https://keycloak.example.com/auth/realms/miq/protocol/openid-connect/token/introspect"
+```
+
+  * The value of **oidc_provider_token_endpoint** varies based on the configuration and the OpenID-Connect Provider being used. 
+
+  * When using IBM's Identiy Access Management (IAM) for the OpenID-Connect Provider, the value for **oidc_provider_token_endpoint** will be the value of OIDCProviderTokenEndpoint from the OpenID-Connect configuration with **token** replace with **identitytoken**.
```
Comment (abellotti) [inline site/_posts/2020-09-03-oidc-auth-api-usage.md:92]: Seems a bit hard to follow telling me replace this with that,  maybe it can be something like: 

```
Each OpenID-Connect Provider defines its own endpoints for token generation and introspection

Example for Keycloak:

    OIDCProviderTokenEndpoint ....
    OIDCOAuthIntrospectionEndpoint ....

    oidc_provider_token_endpoint=https://....
    oidc_auth_introspection_endpoint=https://....

Example for IAM:

    OIDCProviderTokenEndpoint  ....
    OIDCOAuthIntrospectionEndpoint  ....

    oidc_provider_token_endpoint=https://....
    oidc_auth_introspection_endpoint=https://....
```

Comment (jvlcek) [reply site/_posts/2020-09-03-oidc-auth-api-usage.md:92]: Good idea. Will do

File: site/_posts/2020-09-03-oidc-auth-api-usage.md [hunk 5]
```diff
@@ -78,15 +83,43 @@ oidc_client_secret="12345678-1234-1234-1234-01234567abcd"
...
-# OIDCProviderTokenEndpoint from the OpenID-Connect configurationt
-# Sometimes OIDCOauthIntrospectionEndpoint without the trailing "introspect"
-oidc_provider_token_endpoint="https://keycloak.example.com/auth/realms/miq/protocol/openid-connect/token/"
 ```
 
-## Example: Using a JWT Token
+Each OpenID-Connect Provider defines its own endpoints for token generation and introspection. Consequently
+the value of these variables depend on the OpenID-Connect Provider being used.
+
+**Example for IBM's Identiy Access Management (IAM):**
```
Comment (chessbyte) [inline site/_posts/2020-09-03-oidc-auth-api-usage.md:91]: typo: `Identiy`

Comment (jvlcek) [reply site/_posts/2020-09-03-oidc-auth-api-usage.md:91]: @chessbyte Good catch. It's odd the spell checker doesn't flag this.

File: site/_posts/2020-09-03-oidc-auth-api-usage.md [hunk 6]
```diff
@@ -78,15 +83,39 @@ oidc_client_secret="12345678-1234-1234-1234-01234567abcd"
...
+> oidc_auth_introspection_endpoint="https://cp-console.apps.iam.example.com/idprovider/v1/auth/introspect"
+> oidc_provider_token_endpoint="https://cp-console.apps.iam.example.com/idprovider/v1/auth/identitytoken"
+> ```
+
+**Example for Keycloak:**
+
+> Given this configuration:
+> ```bash
+> OIDCOAuthIntrospectionEndpoint https://keycloak.example.com/auth/realms/miq/protocol/openid-connect/token/introspect 
+> ```
```
Comment (abellotti) [inline site/_posts/2020-09-03-oidc-auth-api-usage.md:110]: how come there is no OIDCProviderTokenEndpoint here ?

Comment (jvlcek) [reply site/_posts/2020-09-03-oidc-auth-api-usage.md:110]: @abellotti 

Because keycloak supports OIDCProviderMetadataURL.

see: https://github.com/zmartzone/mod_auth_openidc/blob/master/auth_openidc.conf#L63

Comment (jvlcek) [reply site/_posts/2020-09-03-oidc-auth-api-usage.md:110]: @abellotti Did you just say "oh yeah, now I remeber" ? 😄

Comment (abellotti) [reply site/_posts/2020-09-03-oidc-auth-api-usage.md:110]: oh that !! 😄

**Review by abellotti — APPROVED** [review]
